### ISSUE TITLE
Interface implementation doesn't seem strict enough

### ISSUE THREAD

**Issue by garysoed** [microsoft/TypeScript#43523]
# Bug Report

### 🔎 Search Terms

interface implementation

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about interfaces

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=7&ts=4.2.3#code/JYOwLgpgTgZghgYwgAgCoQM5gJLmQbwChkTkBbCMACwHsATACgDc4AbAVwgC5kR2yARtAA+WKKADmASh5MawOgG5CAX0KEA9BuQA5GsgQ0yAB2CsUYYBWTQoNKBgB0hBKzgYMaTDhOtkV43MKcE90LFwwAmJSCmp6ZjZOHjFJGWQ5BSjSbPTEiEcEKjgoAEEwBgAmAAYpZWy1NRcaECx-HjCcPABeXggAdy9w3wZawmBHWNpGAEYKgGZa5C1kACV2cCsUW3sAQmigA)

### 💻 Code

```ts
interface TestInt {
    method(value: number|string): void;
}

// No compile time errors.
class TestImpl implements TestInt {
    method(value: string): void {
        value.charAt(20);
    }
}

const i: TestInt = new TestImpl();
i.method(123); // Runtime error!
```

### 🙁 Actual behavior

TS compiler doesn't complain that `TestImpl` doesn't implement `TestInt` correctly. This results in runtime error.

### 🙂 Expected behavior

TS compiler should complaing that `TestImpl` doesn't implement `TestInt` correctly. The interface says that `method` can accept in a `number` or a `string`, but the implementation only accepts `string`.

**Comment by whzx5byb** [comment]
Method arguments are always bivariant. See https://github.com/microsoft/TypeScript/pull/18654 and https://github.com/microsoft/TypeScript/wiki/FAQ#why-are-function-parameters-bivariant.

If you want the expected behavior, you should declare it as a "function property" instead of a "method".
```
interface TestInt {
    method: (value: number|string) => void;
}
```

**Comment by garysoed** [comment]
Thanks for the explanation! Closing this.
